src/main/java/org/jenkinsci/plugins/workflow/support/steps/build/BuildTriggerStepExecution.java

```diff
@@ -167,15 +171,31 @@ protected Job asJob() {
                             allParameters.put(defaultP.getName(), defaultP);
                         }
                     } else {
+                        String description = Util.fixNull(pDef.getDescription());
                         if (pDef instanceof ChoiceParameterDefinition) {
                             ParameterValue pv = allParameters.get(pDef.getName());
                             if (!((ChoiceParameterDefinition)pDef).getChoices().contains(pv.getValue())) {
                                 throw new AbortException("Value for choice parameter '" + pDef.getName() + "' is '" + pv.getValue() + "', "
                                         + "but valid choices are " + ((ChoiceParameterDefinition)pDef).getChoices());
                             }
+                        } else if (pDef instanceof PasswordParameterDefinition) {
```

Comment:
Why not just convert every `StringParameterValue` where the `ParameterDefinition` is a subtype of `SimpleParameterDefinition` but not `StringParameterDefinition` using `SimpleParameterDefinition.createValue` as in https://github.com/jenkinsci/parameterized-trigger-plugin/pull/62?

Reply:
I tried your proposal, please look at the last commit. From my PoV it adds complexity due to the number of corner cases.

Feel free to revert the last commit, change it yourself, or merge it like this :)

Reply:
Replied down in https://github.com/jenkinsci/pipeline-build-step-plugin/pull/34#discussion_r333023820, mostly I am just trying to understand the problem better, not saying "do this" or "do that". Thanks for submitting a PR for this!